src/families/bitcoin/wallet-btc/storage/index.ts

```diff
@@ -20,23 +23,35 @@ class BitcoinLikeStorage implements IStorage {
   // returning unordered tx within the same block)
   spentUtxos: { [key: string]: Input[] } = {};
 
-  getLastTx(txFilter: {
-    account?: number;
-    index?: number;
-    address?: string;
-    confirmed?: boolean;
-  }) {
+  getLastTx(txFilter: { account: number; index: number; confirmed?: boolean }) {
+    if (
+      typeof this.accountIndex[`${txFilter.account}-${txFilter.index}`] ===
+      "undefined"
+    ) {
+      return undefined;
+    }
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore
-    const tx: TX | undefined = findLast(this.txs, (t) => {
-      return (
-        (!has(txFilter, "account") || t.account === txFilter.account) &&
-        (!has(txFilter, "index") || t.index === txFilter.index) &&
-        (!has(txFilter, "address") || t.address === txFilter.address) &&
-        (!has(txFilter, "confirmed") ||
+    const tx: TX | undefined = findLast(
+      this.accountIndex[`${txFilter.account}-${txFilter.index}`].map(
```

Comment:
please avoid all the repetition maybe with a `hashAccountIndex(tx)` funciton shared across every parts.